cpp/benchmarks/io/parquet/parquet_reader_input.cpp

```diff
@@ -56,34 +56,30 @@ void parquet_read_common(cudf::io::parquet_writer_options const& write_opts,
   state.add_buffer_size(source_sink.size(), "encoded_file_size", "encoded_file_size");
 }
 
-template <data_type DataType, cudf::io::io_type IOType>
-void BM_parquet_read_data(
-  nvbench::state& state,
-  nvbench::type_list<nvbench::enum_type<DataType>, nvbench::enum_type<IOType>>)
+template <data_type DataType>
+void BM_parquet_read_data(nvbench::state& state, nvbench::type_list<nvbench::enum_type<DataType>>)
 {
-  auto const d_type                 = get_type_or_group(static_cast<int32_t>(DataType));
-  cudf::size_type const cardinality = state.get_int64("cardinality");
-  cudf::size_type const run_length  = state.get_int64("run_length");
-  auto const compression            = cudf::io::compression_type::SNAPPY;
+  auto const d_type      = get_type_or_group(static_cast<int32_t>(DataType));
+  auto const cardinality = static_cast<cudf::size_type>(state.get_int64("cardinality"));
+  auto const run_length  = static_cast<cudf::size_type>(state.get_int64("run_length"));
+  cudf::io::io_type const source_type = retrieve_io_type_enum(state.get_string("io_type"));
```

Comment:
Thank you, @ttnghia! I can update the PR. I also wanted to ask if it's a general practice to use implicit type inference wherever possible. Generally, I find code more readable when explicit types are used.

Reply:
It is recommended to use `auto const` for modern C++. Yes explicit type is more readable, but it is not recommended because it is more error-prone: You may put in a wrong type, or you change the function return type but forgot/don't know to change the type at its caller.

Reply:
Thanks! I'll keep that in mind!